test/WireMock.Net.Tests/WireMockServerTests.cs

```diff
@@ -196,6 +196,72 @@ public async Task WireMockServer_When_HttpClientWithWebProxyCallsHttp_Should_Wor
     }
 #endif
 
+[Fact]
+public async Task WireMockServer_WithUrl0000_Should_Listen_On_All_IPs()
+{
+    // Arrange
+
+    List<string> IPv4 = new List<string>();
+    List<string> IPv6 = new List<string>();
+
+    foreach (NetworkInterface netInterface in NetworkInterface.GetAllNetworkInterfaces())
+    {
+        if (netInterface.OperationalStatus == OperationalStatus.Up)
+        {
+
+            IPInterfaceProperties ipProps = netInterface.GetIPProperties();
+
+            foreach (UnicastIPAddressInformation addr in ipProps.UnicastAddresses)
+            {
+                Console.WriteLine(" " + addr.Address.ToString());
+
+                if (addr.Address.AddressFamily == System.Net.Sockets.AddressFamily.InterNetwork)
+                {
+                    IPv4.Add(addr.Address.ToString());
+                }
+                else if (addr.Address.AddressFamily == System.Net.Sockets.AddressFamily.InterNetworkV6)
+                {
+                    IPv6.Add($"[{addr.Address.ToString()}]");
+                }
+            }
+        }
+    }
+
+    // Asser.
+    IPv4.Should().NotBeEmpty();
```

Comment:
It's not 100% sure that the system which is running has ip-addresses allocated, so better remove this check.

Reply:
I thought a test should fail, because it cannot test if it has no IPv4 / IPv6 in this case? Would false assume the test is OK but the test is not OK if it cannot test?

Reply:
Additionally Firewalls could block access. 
Could be surrounded with try catch and warn?

Reply:
OK.
Make 2 tests.

And it seems that the unit tests fail on Linux:
```
[xUnit.net 00:00:39.60]     WireMock.Net.Tests.WireMockServerTests.WireMockServer_WithUrl0000_Should_Listen_On_All_IPs [FAIL]
  Failed WireMock.Net.Tests.WireMockServerTests.WireMockServer_WithUrl0000_Should_Listen_On_All_IPs [10 s]
  Error Message:
   WireMock.Exceptions.WireMockException : Service start failed with error: One or more errors occurred. (Permission denied)
---- System.AggregateException : One or more errors occurred. (Permission denied)
-------- System.Net.Sockets.SocketException : Permission denied
```

And Windows:
```
[xUnit.net 00:00:36.61]     WireMock.Net.Tests.WireMockServerTests.WireMockServer_WithUrl0000_Should_Listen_On_All_IPs [FAIL]
  Failed WireMock.Net.Tests.WireMockServerTests.WireMockServer_WithUrl0000_Should_Listen_On_All_IPs [10 s]
  Error Message:
   WireMock.Exceptions.WireMockException : Service start failed with error: One or more errors occurred. (An attempt was made to access a socket in a way forbidden by its access permissions.)
---- System.AggregateException : One or more errors occurred. (An attempt was made to access a socket in a way forbidden by its access permissions.)
-------- System.Net.Sockets.SocketException : An attempt was made to access a socket in a way forbidden by its access permissions.
```

So probably you need to add a new helper class to read the `TF_BUILD` environment variable --> indicates that the tests are running in devops ci-cd.

And make sure to merge the latest main into your branch, I made some fixes in the CI.

Reply:
I added a custom fact type that could be reused and could be used if other types are needed in future (like excluding if Linux)?